### ISSUE TITLE
[Feat]: Ability to extract metrics (for example for Spring based apps) easily by making sure the Requestor is pluggable

### ISSUE THREAD

**Issue by nandorholozsnyak** [EasyPost/easypost-java#321]
### Feature Request Is New

- [X] I have verified that the requested feature does not already exist or has not already been requested.

### Description of the feature

Hello there,

I'm looking forward for a way to create metrics about the API calls towards to the EasyPost API.

With the SDK it seemed like we would be able to use the Request and Response hooks, but based on the codebase, response hooks are not initiated if the response contains any kind of error (more specifically if this is true: 
```java
if (rCode < HttpURLConnection.HTTP_OK || rCode >= HttpURLConnection.HTTP_MULT_CHOICE) {
    handleAPIError(rBody, rCode);
}
```

I tried to plug Micrometer a layer below into the `com.easypost.http.Requestor#createEasyPostConnection` method, but it is still not the best place, as we need a custom URLStreamHandler which still not the best solution. 
I like how it is almost fully dependency free, but at the same time it is a painpoint to use the low level HttpsURLConnection.

So my question is, what is the best way to capture the start and the end of any API call to have the response code, URL and basically everything that I would need for a standard `http.client.requests` metrics which includes these tags/labels (http_method, http_status_code, outcome,error,uri etc.).

**Comment by nwithan8** [comment]
Hello, thanks for reaching out!

We can look into wiring up the response hooks to fire even if an HTTP error occurs, so you can capture and assess the response.

**Comment by nandorholozsnyak** [comment]
> Hello, thanks for reaching out!
> 
> We can look into wiring up the response hooks to fire even if an HTTP error occurs, so you can capture and assess the response.

Hello there,

thank you for the quick answer, I could understand if it was made on purpose for not running those response hooks, but at the same time it could help us.
If there are other integration/injection points to intercept these kind of data, I'm open for any of that.

**Comment by nwithan8** [comment]
We have released an update that should allow the response hooks to fire even if there is an API error, available in [v7.4.1](https://github.com/EasyPost/easypost-java/releases/tag/v7.4.1)

**Comment by nandorholozsnyak** [comment]
> We have released an update that should allow the response hooks to fire even if there is an API error, available in [v7.4.1](https://github.com/EasyPost/easypost-java/releases/tag/v7.4.1)

That was fast. Thank you.
